### PR TITLE
❇️  Add Convert `patches` Output to `AnnotationStore`

### PR DESCRIPTION
**Pull request by measty**
Pulls out the changes related to storing patches in an AnnotationStore from dev-redefine-patchpredictor into a separate PR.

Implements a function `patch_pred_store` to convert the output from PatchPredictor into an AnnotationStore.

### CODE REVIEW

File: tiatoolbox/utils/misc.py [hunk 1]
```diff
@@ -1194,3 +1195,65 @@ def add_from_dat(
...
+    """
+    if "coordinates" not in patch_output:
+        # we cant create annotations without coordinates
+        msg = "Patch output must contain coordinates."
+        raise ValueError(msg)
+    # get relevant keys
+    class_probs = patch_output.get("probabilities", [])
+    preds = patch_output.get("predictions", [])
+    patch_coords = np.array(patch_output.get("coordinates", []))
+    if not np.all(scale_factor == 1):
```
Comment (shaneahmed) [inline tiatoolbox/utils/misc.py:1227]: Are you able to cover this line in tests.

File: tiatoolbox/utils/misc.py [hunk 2]
```diff
@@ -1194,3 +1195,65 @@ def add_from_dat(
...
+            keys are "probabilities", "predictions", "coordinates", and "labels".
+        scale_factor (tuple[int, int]): The scale factor to use when loading the
+            annotations. All coordinates will be multiplied by this factor to allow
+            conversion of annotations saved at non-baseline resolution to baseline.
+            Should be model_mpp/slide_mpp.
+        class_dict (dict): Optional dictionary mapping class indices to class names.
+
+    Returns:
+        SQLiteStore: An SQLiteStore containing Annotations for each patch.
+    """
```
Comment (shaneahmed) [inline tiatoolbox/utils/misc.py:1218]:
```suggestion

    """
```

**Review by shaneahmed — COMMENTED** [review]
Just left a couple of comments. Otherwise it looks good.

File: tiatoolbox/utils/misc.py (hunk 1 above)
Comment (measty) [reply tiatoolbox/utils/misc.py]: done

**Review by shaneahmed — APPROVED** [review]
Thanks @measty 

@AbishekRajVG Please let me know once you have approved this PR. So I can merge it then.

**Review by AbishekRajVG — APPROVED** [review]

**Pull request merged**